### PR TITLE
:seedling: Add test case to descendantsTestCases

### PR DESCRIPTION
**Pull request by zhouhao3**
Improve unit test coverage:

CountDescendants: 75% -> 100% coverage
listDescendants: 72.7% -> 77.8% coverage

Signed-off-by: Zhou Hao <zhouhao@fujitsu.com>

### CODE REVIEW

File: baremetal/metal3cluster_manager_test.go [hunk 1]
```diff
@@ -316,9 +318,29 @@ var _ = Describe("Metal3Cluster manager", func() {
 					},
 				},
 			},
+			OwnerRef:            bmcOwnerRef,
 			ExpectError:         false,
 			ExpectedDescendants: 1,
 		}),
+		Entry("One Cluster Descendant with error OwmRef", descendantsTestCase{
```
Comment (furkatgofurov7) [inline baremetal/metal3cluster_manager_test.go:325]: we could improve this message a bit, i.e 
```suggestion
		Entry("One Cluster Descendant with wrong OwnerRef", descendantsTestCase{
```

and in the previous entry to make the difference a bit clear:
```suggestion
		Entry("One Cluster Descendant with proper OwnerRef", descendantsTestCase{
```

**Review by furkatgofurov7 — COMMENTED** [review]
One suggestion inline, otherwise looks good to me:

File: baremetal/metal3cluster_manager_test.go (hunk 1 above)
Comment (zhouhao3) [reply baremetal/metal3cluster_manager_test.go]: Updated, PTAL.

File: baremetal/metal3cluster_manager_test.go [hunk 2]
```diff
@@ -302,10 +303,11 @@ var _ = Describe("Metal3Cluster manager", func() {
 	var descendantsTestCases = []TableEntry{
 		Entry("No Cluster Descendants", descendantsTestCase{
 			Machines:            []*clusterv1.Machine{},
+			OwnerRef:            bmcOwnerRef,
 			ExpectError:         false,
 			ExpectedDescendants: 0,
 		}),
-		Entry("One Cluster Descendant", descendantsTestCase{
+		Entry("One Cluster Descendant with proper OwmRef", descendantsTestCase{
```
Comment (furkatgofurov7) [inline baremetal/metal3cluster_manager_test.go:310]: typo: OwnerRef

File: baremetal/metal3cluster_manager_test.go [hunk 3]
```diff
@@ -316,9 +318,29 @@ var _ = Describe("Metal3Cluster manager", func() {
 					},
 				},
 			},
+			OwnerRef:            bmcOwnerRef,
 			ExpectError:         false,
 			ExpectedDescendants: 1,
 		}),
+		Entry("One Cluster Descendant with wrong OwmRef", descendantsTestCase{
```
Comment (furkatgofurov7) [inline baremetal/metal3cluster_manager_test.go:325]: same

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

File: baremetal/metal3cluster_manager_test.go (hunk 2 above)
Comment (zhouhao3) [reply baremetal/metal3cluster_manager_test.go]: Updated, thanks.

**Review by kashifest — COMMENTED** [review]
/lgtm

**Review by furkatgofurov7 — APPROVED** [review]
Thanks @zhouhao3!

/approve

**Pull request merged**